src/components/course/data/hooks.js

```diff
@@ -254,3 +263,80 @@ export function useCourseEnrollmentUrl({
 
   return enrollmentUrl;
 }
+
+/**
+ * A hook to obtain data, for use with the enroll button for example. It answers these questions
+ * by reading the CourseContext context:
+ *
+ * 1. isUserEnrolled
+ * 2. isCourseStarted
+ * 3. isEnrollable
+ *
+ * These three answers can help with the main flow of EnrollButton.
+ */
+export function useCourseData() {
+  const { state: courseData } = useContext(CourseContext);
```

Comment:
Does CourseContext call an API? If yes, I think I'd like to make sure we're not calling it twice by having it in each hook. Maybe put the `useContext` outside the hook and pass it in?